packages/s3-store/index.ts

```diff
@@ -197,6 +197,21 @@ export class S3Store extends DataStore {
     }
   }
 
+  private async getIncompletePartSize(id: string): Promise<number | undefined> {
+    try {
+      const data = await this.client.headObject({
+        Bucket: this.bucket,
+        Key: this.partKey(id, true),
+      })
+      return data.ContentLength
+    } catch (error) {
+      if (error instanceof NotFound) {
+        return undefined
```

Comment:
nit: when there is no incomplete part when can also consider the size `0`. Especially if we want to always do addition on the return value.

```suggestion
        return 0
```

Reply:
I was thinking about that. In this case we would not be able to tell if there is no incomplete part or if its size is `0`. For HEAD request that distinguishing is actually not needed. I'm good with both ways.